python/test/core/test_memcontent.py

```diff
@@ -0,0 +1,412 @@
+        addr = 0x1234
+        data = bytes(range(10))
+        memcontent.write(addr, data)
+
+        with self.assertRaises(Exception):
+            memcontent.read(addr - 1, len(data))
+
+        with self.assertRaises(Exception):
+            memcontent.read(addr + 1, len(data))
+
+    def test_merge_write(self):
+        memcontent = MemContent()
+        data = bytes(range(10))
+        memcontent.write(0x1000, data)
+        memcontent.write(0x1005, data)
+        data2 = memcontent.read(0x1000, 15)
+        self.assertEqual(data[0:5] + data, data2)
+
+    def test_merge_write_limit_low_left(self):
+        memcontent = MemContent()
+        data = bytes(range(10))
+        memcontent.write(1000, data)
+        memcontent.write(990, data)
+        data2 = memcontent.read(990, 20)
+        self.assertEqual(data + data, data2)
+
+    def test_merge_write_limit_high(self):
+        memcontent = MemContent()
+        data = bytes(range(10))
+        memcontent.write(990, data)
+        memcontent.write(1000, data)
+        data2 = memcontent.read(990, 20)
+        self.assertEqual(data + data, data2)
+
+    def test_merge_write_middle(self):
+        memcontent = MemContent()
+        data1 = bytes(range(30))
+        data2 = bytes(range(10))
+        memcontent.write(1000, data1)
+        memcontent.write(1010, data2)
+        data3 = memcontent.read(1000, 30)
+        self.assertEqual(data1[0:10] + data2 + data1[20:30], data3)
+
+    def test_write_mutiple_overlap(self):
+        memcontent = MemContent()
+        data = bytes(range(10))
+        memcontent.write(990, data)
+        memcontent.write(1000, data)
+        memcontent.write(1005, data)
+        memcontent.write(800, data)
+        memcontent.write(1100, data)
+        memcontent.write(995, data)
+        self.assertEqual(data, memcontent.read(800, len(data)))
+        self.assertEqual(data, memcontent.read(1100, len(data)))
+        data2 = memcontent.read(990, 25)
+        self.assertEqual(data[0:5] + data + data, data2)
+
+    def test_cluster_definition(self):
+        memcontent = MemContent()
+        data = bytes(range(10))
+        memcontent.write(990, data)
+        memcontent.write(1000, data)
+        memcontent.write(1005, data)
+        memcontent.write(800, data)
+        memcontent.write(1100, data)
+        memcontent.write(995, data)
+
+        clusters = memcontent.get_cluster_list_no_data()  # Clusters should be sorted by address
+
+        self.assert_clusters(clusters, [(800, 10), (990, 25), (1100, 10)])
+
+    # ==== Deletion Test =======
+
+    def delete_test_make_mem(self):
+        memcontent = MemContent()
+        data = bytes(range(0x100))
+        memcontent.write(0x1000, data)
+        memcontent.write(0x2000, data)
+        memcontent.write(0x3000, data)
+
+        return memcontent
+
+    def test_delete_single_cluster_middle(self):
+        import ipdb
+        m = self.delete_test_make_mem()
+        # ipdb.set_trace()
```

Comment:
Cleanup developper support imports statement `import ipdb` and ipdb related comments.

Reply:
Fixed